scripts/bump-version.js

```diff
@@ -8,43 +8,87 @@
 const path = require('path')
+const {saveJSONToFile, setPackageVersion} = require('./utils')
 
+// Exit upon error
 sh.set('-e')
+
 const lernaConfigPath = path.join(__dirname, '..', 'lerna.json')
-const rootPkgPath = path.join(__dirname, '..', 'package.json')
-const rootPkgLockPath = path.join(__dirname, '..', 'package-lock.json')
+const rootPath = path.join(__dirname, '..')
+
+const monorepoPackages = JSON.parse(sh.exec('lerna list --all --json', {silent: true}))
+const monorepoPackageNames = monorepoPackages.map((pkg) => pkg.name)
+
+const INDEPENDENT_PACKAGES = JSON.parse(
+    sh.cat(path.join(__dirname, 'packages-with-independent-version.json'))
+)
+const independentPackages = INDEPENDENT_PACKAGES.map((pkgName) =>
+    monorepoPackages.find((pkg) => pkg.name === pkgName)
+)
 
 const main = () => {
-    sh.exec(`lerna version --no-push --no-git-tag-version --yes ${process.argv.slice(2).join(' ')}`)
-    sh.exec(`npm install`)
+    const args = process.argv.slice(2).join(' ')
+    sh.exec(`lerna version --exact --no-push --no-git-tag-version --yes ${args}`)
+    // `--exact` above is for pinning the version of the pwa-kit dependencies
+    // https://github.com/lerna/lerna/tree/main/libs/commands/version#--exact
+
     const lernaConfig = JSON.parse(sh.cat(lernaConfigPath))
-    const rootPkg = JSON.parse(sh.cat(rootPkgPath))
-    const rootLockPkg = JSON.parse(sh.cat(rootPkgLockPath))
-
-    // find all monorepo packages, look inside each package json, find peerDependency that is a monorepo package
-    // and update it with a new version
-    const {stdout} = sh.exec('lerna list --all --json', {silent: true})
-    const packages = JSON.parse(stdout.toString())
-    const lernaPackageNames = packages.map((pkg) => pkg.name)
-    packages.forEach(({location}) => {
-        const pkgFilePath = path.join(location, 'package.json')
-        const pkg = JSON.parse(sh.cat(pkgFilePath))
-        const peerDependencies = pkg.peerDependencies
-        if (!peerDependencies) return
-        Object.keys(peerDependencies).forEach((dep) => {
-            if (lernaPackageNames.includes(dep)) {
-                console.log(`Found lerna local package ${dep} as a peer dependency of ${pkg.name}.`)
-                peerDependencies[dep] = `^${lernaConfig.version}`
-                new sh.ShellString(JSON.stringify(pkg, null, 2)).to(pkgFilePath)
-            }
-        })
-    })
+    const newMonorepoVersion = lernaConfig.version
 
     // update versions for root package and root package lock
-    rootPkg.version = lernaConfig.version
-    rootLockPkg.version = lernaConfig.version
-    new sh.ShellString(JSON.stringify(rootPkg, null, 2)).to(rootPkgPath)
-    new sh.ShellString(JSON.stringify(rootLockPkg, null, 2)).to(rootPkgLockPath)
+    setPackageVersion(newMonorepoVersion, {cwd: rootPath})
+
+    independentPackages.forEach((pkg) => {
+        const {location, version: oldVersion} = pkg
+        // Restore and then increment to the next pre-release version
```

Comment:
Is there a particular reason why we aren't leveraging lerna's [independent](https://lerna.js.org/docs/features/version-and-publish#independent-mode) mode? That way you don't need to do the "increment then restore" dance.

Reply:
@kevinxh What we're aiming for is a hybrid setup that Lerna does not provide. It does things all or nothing: either all packages in the monorepo have fixed or independent version.

Reply:
After discussing further, we both wonder whether we can turn everything into independent version (as implementation detail). The interface would still be the same.

Reply:
@kevinxh if we turn everything into independent version, unfortunately Lerna does not allow us to programatically set the versions via the CLI.

For example:
- if `lerna version`, then it will go into interactive mode and ask us to enter in the version for each package in the monorepo. This is tedious and can be error-prone.
- if `lerna version 3.0.0`, then it will attempt to set the versions of all packages into 3.0.0.

I wish the CLI can accept multiple versions and associate them with different packages. That wish is echoed by others too in this feature request: https://github.com/lerna/lerna/issues/389